tf2_ros_py/tf2_ros/buffer.py

```diff
@@ -338,13 +338,13 @@ def wait_for_transform_full_async(
         fut = rclpy.task.Future()
         if self.can_transform_full_core(target_frame, target_time, source_frame, source_time, fixed_frame)[0]:
             # Short cut, the transform is available
-            fut.set_result(True)
+            fut.set_result(self.lookup_transform(target_frame, source_frame, time))
```

Comment:
Oh, right.  This should probably be `self.lookup_transform_full_core`, to match the `can_transform_full_core` above.